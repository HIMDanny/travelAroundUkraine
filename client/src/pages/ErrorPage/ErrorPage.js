import React from 'react';
import { Box, Typography, Button, Container } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import styled from '@emotion/styled';

const ErrorWrapper = styled('div')({
  marginTop: '90px',
  marginBottom: '150px',
  width: '100%',
  height: '100%',
  textAlign: 'center',
  '& > div': {
    display: 'flex',
    justifyContent: 'center',
    '& > img': {
      width: '50vw',
      maxWidth: '500px',
    },
  },
});

const ErrorPage = () => {
  const navigate = useNavigate();
  return (
    <Container>
      <ErrorWrapper>
        <Box>
          <Box src="/assets/images/404/heart.png" component="img" alt="uk-flag" />
        </Box>
        <Typography sx={{ mb: '40px' }} variant="h2">
          404 page not found
        </Typography>
        <Typography>Please, navigate to main page to plan your trip</Typography>
        <Box sx={{ mt: '40px'}}>
          <Button onClick={() => navigate('/')}>Home</Button>
        </Box>
      </ErrorWrapper>
    </Container>
  );
};

export default ErrorPage;
