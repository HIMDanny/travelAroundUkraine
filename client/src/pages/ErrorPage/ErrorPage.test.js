/* eslint-disable react/button-has-type */
/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ErrorPage from './ErrorPage';

const RouterWrapper = ({ children }) => <MemoryRouter>{children}</MemoryRouter>;
jest.mock('@mui/material/Box', () => 'div');
jest.mock('@mui/material/Container', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);

describe('Error page snapshot testing', () => {
  test('should Error page renders', () => {
    const { asFragment } = render(<ErrorPage />, { wrapper: RouterWrapper });
    expect(asFragment()).toMatchSnapshot();
  });
});
