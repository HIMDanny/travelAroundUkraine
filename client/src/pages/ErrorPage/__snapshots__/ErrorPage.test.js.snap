// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page snapshot testing should Error page renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="css-j67klr"
    >
      <div>
        <div
          alt="uk-flag"
          component="img"
          src="/assets/images/404/heart.png"
        />
      </div>
      <p>
        404 page not found
      </p>
      <p>
        Please, navigate to main page to plan your trip
      </p>
      <div
        sx="[object Object]"
      >
        <button>
          Home
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
